src/transporters/amqp10.js

```diff
@@ -0,0 +1,456 @@
+/*
+ * moleculer
+ * Copyright (c) 2019 MoleculerJS (https://github.com/moleculerjs/moleculer)
+ * MIT Licensed
+ */
+
+"use strict";
+
+const url = require("url");
+const Promise = require("bluebird");
+const Transporter = require("./base");
+const { isPromise } = require("../utils");
+
+const {
+	PACKET_REQUEST,
+	PACKET_RESPONSE,
+	PACKET_UNKNOWN,
+	PACKET_EVENT,
+	PACKET_DISCOVER,
+	PACKET_INFO,
+	PACKET_DISCONNECT,
+	PACKET_HEARTBEAT,
+	PACKET_PING,
+	PACKET_PONG
+} = require("../packets");
+
+/**
+ * Transporter for AMQP 1.0
+ *
+ * More info: https://www.amqp.org/resources/specifications
+ *
+ * @class Amqp10Transporter
+ * @extends {Transporter}
+ */
+class Amqp10Transporter extends Transporter {
+	/**
+	 * Creates an instance of Amqp10Transporter.
+	 *
+	 * @param {any} opts
+	 *
+	 * @memberof Amqp10Transporter
+	 */
+	constructor(opts) {
+		if (typeof opts == "string") opts = { url: opts };
+
+		super(opts);
+
+		/* istanbul ignore next*/
+		if (!this.opts) this.opts = {};
+
+		// Number of requests a broker will handle concurrently
+		if (typeof opts.prefetch !== "number") opts.prefetch = 1;
+
+		// Number of milliseconds before an event expires
+		if (typeof opts.eventTimeToLive !== "number") opts.eventTimeToLive = null;
+
+		if (typeof opts.heartbeatTimeToLive !== "number") opts.heartbeatTimeToLive = null;
+
+		if (typeof opts.queueOptions !== "object") opts.queueOptions = {};
+
+		if (typeof opts.messageOptions !== "object") opts.messageOptions = {};
+
+		this.receivers = [];
+		this.hasBuiltInBalancer = true;
+		this.connection = null;
+	}
+
+	_getQueueOptions(packetType, balancedQueue) {
+		let packetOptions = {};
+		switch (packetType) {
+			// Requests and responses don't expire.
+			case PACKET_REQUEST:
+				// TODO: auto delete
+				break;
+			case PACKET_RESPONSE:
+				// TODO: auto delete
+				break;
+
+			// Consumers can decide how long events live
+			// Load-balanced/grouped events
+			case PACKET_EVENT + "LB":
+			case PACKET_EVENT:
+				// TODO: auto delete
+				break;
+
+			// Packet types meant for internal use
+			case PACKET_HEARTBEAT:
+				// TODO: auto delete
+				// packetOptions = {};
+				break;
+			case PACKET_DISCOVER:
+			case PACKET_DISCONNECT:
+			case PACKET_UNKNOWN:
+			case PACKET_INFO:
+			case PACKET_PING:
+			case PACKET_PONG:
+				// TODO: auto delete
+				break;
```

Comment:
Due to many ToDos can you explain a little bit more about what's missing?

Reply:
we didn't find any naive `autoDeleteQueues` option implementation, so we left it as a `TODO`. anyways, you could always just overwrite `opts.queueOptions` or `opts.topicOptions` to your liking.